extensions/VisualEditor/wikia/ApiAddMediaTemporary.php

```diff
@@ -154,6 +156,14 @@ protected function checkPermissions() {
 		}
 	}
 
+	protected function checkPermissionsForJA() {
```

Comment:
Again, Japan is special case that we allow anonymous file upload. So even if I follow your suggestion, then I still have to do: if (local is japan) then allow anonymous upload if not wgdisableanonymousediting (pseudo code). 

Rather than if.. statement woven into the existing function ( checkPermissions ), it is a lot clear to have a separate function, in my opinion. But you think opposite, that is interesting.